common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java

```diff
@@ -499,4 +499,264 @@ public void onTaskCompleted(Void aVoid) {
+
+        mTestCacheFile.putString(key, legacyEncryptedValue);
+
+        // Attempt to reencrypt the cache, but provide the wrong key intentionally
+        mFileManagerReencrypter.reencrypt(
+                mTestCacheFile,
+                mTestEncrypterDecrypter,
+                mTestEncrypterDecrypter,
+                new ISharedPrefsFileManagerReencrypter.ReencryptionParams(
+                        false,
+                        false,
+                        false
+                )
+        );
+
+        // Assert nothing was done
+        Assert.assertEquals(legacyEncryptedValue, mTestCacheFile.getString(key));
+    }
+
+    @Test(expected = Exception.class)
+    public void testIncorrectKeyProvidedThrows() throws Exception {
+        final byte[] mockKey = generateLegacyFormatKey("abcdefabcd");
+        final TestEncrypterDecrypter origDelegate = new TestEncrypterDecrypter(mContext, mockKey);
+
+        final String plaintextValue = "a_cool_value";
+
+        final String key = "key";
+        final String legacyEncryptedValue = origDelegate.encryptWithLegacyKey(plaintextValue);
+
+        mTestCacheFile.putString(key, legacyEncryptedValue);
+
+        // Attempt to reencrypt the cache, but provide the wrong key intentionally
+        mFileManagerReencrypter.reencrypt(
+                mTestCacheFile,
+                mTestEncrypterDecrypter,
+                mTestEncrypterDecrypter,
+                new ISharedPrefsFileManagerReencrypter.ReencryptionParams(
+                        true,
+                        false,
+                        false
+                )
+        );
+    }
+
+    @Test
+    public void testIncorrectKeyProvidedClearsEntryMultiple() throws Exception {
+        final byte[] mockKey = generateLegacyFormatKey("abcdefabcd");
+        final TestEncrypterDecrypter origDelegate = new TestEncrypterDecrypter(mContext, mockKey);
+
+        final String plainTextValue = "a_cool_value";
+        final String anotherPlaintextValue = "a_cold_value";
+
+        final String keyOne = "key_one";
+        final String legacyEncryptedValue = origDelegate.encryptWithLegacyKey(plainTextValue);
+
+        final String keyTwo = "key_two";
+        final String anotherLegacyEncryptedValue = origDelegate.encryptWithLegacyKey(anotherPlaintextValue);
+
+        mTestCacheFile.putString(keyOne, legacyEncryptedValue);
+        mTestCacheFile.putString(keyTwo, anotherLegacyEncryptedValue);
+
+        // Attempt to reencrypt the cache, but provide the wrong key intentionally
+        mFileManagerReencrypter.reencrypt(
+                mTestCacheFile,
+                mTestEncrypterDecrypter,
+                mTestEncrypterDecrypter,
+                new ISharedPrefsFileManagerReencrypter.ReencryptionParams(
+                        false,
+                        true,
+                        false
+                )
+        );
+
+        // Assert nothing was done
```

Comment:
That's not what was happening...

Reply:
ea702c7b41052c5c8d4b5428971ea10f1f059d7d